### PR TITLE
Update the why ballerina page for concurrency

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
$title.

Content based mostly on the spec and the language guide.

I haven't added transactions-related content yet. Need to finalize.

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: learn/user-guide/why-ballerina/concurrent.md [hunk 1]
```diff
@@ -5,48 +5,134 @@ description: Concurrency in Ballerina is enabled by strands, which are lightweig
...
+    int id;
+    string location?;
+|};
 
-This concurrency execution is visualized in the generated sequence diagram below.
+function process(map<Details> & readonly entities) {
+    string[] detailsWithLocation = [];
+    string[] detailsWithoutLocation = [];
 
-<img src="/img/why-pages/sequence-diagrams-for-programming-5.png" alt="Ballerina sequence diagram visualizing concurrency">
```
Comment (MaryamZi) [inline learn/user-guide/why-ballerina/concurrent.md:47]: I don't think we can add a diagram atm.

File: learn/user-guide/why-ballerina/concurrent.md [hunk 2]
```diff
@@ -5,48 +5,134 @@ description: Concurrency in Ballerina is enabled by strands, which are lightweig
...
 permalink: /learn/user-guide/why-ballerina/concurrent/
 active: concurrent
-intro: Concurrency in Ballerina is enabled by strands, which are lightweight threads. 
+intro: Ballerina's concurrency model supports both threads and coroutines.
 redirct_from:
   - /learn/user-guide/why-ballerina/concurrent
 ---
 
-A single operating system thread can contain multiple strands. A single strand is run at a time in a thread and the strands belonging to a single thread are cooperatively multitasked. Strands allow you to optimize the usage of CPU time and is beneficial in implementing non-blocking I/O operations.
+Ballerina's concurrency model supports both threads and coroutines, and has been designed to have a close correspondence with sequence diagrams.
```
Comment (praneesha) [inline learn/user-guide/why-ballerina/concurrent.md:13]:
```suggestion
Ballerina's concurrency model supports both threads and coroutines and has been designed to have a close correspondence with sequence diagrams.
```

File: learn/user-guide/why-ballerina/concurrent.md [hunk 3]
```diff
@@ -5,48 +5,134 @@ description: Concurrency in Ballerina is enabled by strands, which are lightweig
...
 
-A worker represents a single strand execution in a function. Concurrent operations in a function are defined by multiple workers as shown in the example below.
+A Ballerina program is executed on one or more threads. A thread can consist of one or more strands, which are language-managed logical threads of control. Only one strand of a particular thread can run at a time. The strands belonging to a single thread are cooperatively multitasked. Therefore, strands of a single thread behave as coroutines relative to each other.
```
Comment (praneesha) [inline learn/user-guide/why-ballerina/concurrent.md:15]:
```suggestion
A Ballerina program is executed on one or more threads. A thread can consist of one or more strands, which are language-managed, logical threads of control. Only one strand of a particular thread can run at a time. The strands belonging to a single thread are cooperatively multitasked. Therefore, strands of a single thread behave as coroutines relative to each other.
```

File: learn/user-guide/why-ballerina/concurrent.md [hunk 4]
```diff
@@ -5,48 +5,134 @@ description: Concurrency in Ballerina is enabled by strands, which are lightweig
...
+
+Strands enable cooperative multitasking by "yielding". The runtime scheduler may switch the execution of a strand only when it yields via a wait action, a worker receive action, a library function resulting in a blocking call, etc.
```
Comment (praneesha) [inline learn/user-guide/why-ballerina/concurrent.md:17]:
```suggestion
Strands enable cooperative multitasking by "yielding". The runtime scheduler may switch the execution of a strand only when it yields via a `wait` action, a `worker-receive` action, a library function resulting in a blocking call, etc.
```

File: learn/user-guide/why-ballerina/concurrent.md [hunk 5]
```diff
@@ -5,48 +5,134 @@ description: Concurrency in Ballerina is enabled by strands, which are lightweig
...
+
+A strand is created due to the execution of either a named worker declaration or start action.
```
Comment (praneesha) [inline learn/user-guide/why-ballerina/concurrent.md:19]:
```suggestion
A strand is created due to the execution of either a named worker declaration or `start` action.
```

File: learn/user-guide/why-ballerina/concurrent.md (hunk 4 above)
Comment (MaryamZi) [reply learn/user-guide/why-ballerina/concurrent.md:17]: What is our criteria for emphasizing using backticks? Wait is a keyword but worker-receive is not.

Comment (praneesha) [reply learn/user-guide/why-ballerina/concurrent.md:17]: Only keywords should be within backticks.

File: learn/user-guide/why-ballerina/concurrent.md (hunk 1 above)
Comment (praneesha) [reply learn/user-guide/why-ballerina/concurrent.md:47]: Let's remove it.

File: learn/user-guide/why-ballerina/concurrent.md (hunk 4 above)
Comment (MaryamZi) [reply learn/user-guide/why-ballerina/concurrent.md:17]: worker-receive isn't a keyword, so will change this to

```suggestion
Strands enable cooperative multitasking by "yielding". The runtime scheduler may switch the execution of a strand only when it yields via a `wait` action, a worker receive action, a library function resulting in a blocking call, etc.
```

Comment (praneesha) [reply learn/user-guide/why-ballerina/concurrent.md:17]: +1

**Review by praneesha — APPROVED** [review]

**Pull request merged**